fdbserver/workloads/BackupCorrectness.actor.cpp

```diff
@@ -517,23 +522,18 @@ struct BackupAndRestoreCorrectnessWorkload : TestWorkload {
 		return Void();
 	}
 
-	ACTOR static Future<Void> clearAndRestoreSystemKeys(Database cx,
-	                                                    BackupAndRestoreCorrectnessWorkload* self,
-	                                                    FileBackupAgent* backupAgent,
-	                                                    Version targetVersion,
-	                                                    Reference<IBackupContainer> lastBackupContainer,
-	                                                    Standalone<VectorRef<KeyRangeRef>> systemRestoreRanges) {
+	ACTOR static Future<Void> restoreSystemKeys(Database cx,
+	                                            Database cxOrig,
+	                                            BackupAndRestoreCorrectnessWorkload* self,
+	                                            FileBackupAgent* backupAgent,
+	                                            Version targetVersion,
+	                                            Reference<IBackupContainer> lastBackupContainer,
+	                                            Standalone<VectorRef<KeyRangeRef>> systemRestoreRanges) {
 		// restore system keys before restoring any other ranges
-		wait(runRYWTransaction(cx, [=](Reference<ReadYourWritesTransaction> tr) -> Future<Void> {
-			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
-			for (auto& range : systemRestoreRanges)
-				tr->clear(range);
-			return Void();
-		}));
 		state Standalone<StringRef> restoreTag(self->backupTag.toString() + "_system");
 		printf("BackupCorrectness, backupAgent.restore is called for tag:%s\n", restoreTag.toString().c_str());
 		wait(success(backupAgent->restore(cx,
-		                                  cx,
+		                                  cxOrig,
```

Comment:
I think we shouldn't change the default behaviour (i.e when encryption is disabled)

Reply:
I'm not sure what this second parameter actually do, but looks like it expected we pass the original Database the backup comes from.

Reply:
Interesting, shouldn't the test would have failed if the requirement was to pass "original database"; maybe the test isn't doing the right thing?